server/src/main/java/io/crate/analyze/Analyzer.java

```diff
@@ -733,5 +747,80 @@ public AnalyzedStatement visitFetch(Fetch fetch, Analysis context) {
             Cursor cursor = context.cursors().get(fetch.cursorName());
             return new AnalyzedFetch(fetch, cursor);
         }
+
+        @Override
+        public AnalyzedStatement visitCreateServer(CreateServer createServer, Analysis context) {
+            ExpressionAnalyzer expressionAnalyzer = new ExpressionAnalyzer(
+                context.transactionContext(),
+                nodeCtx,
+                context.paramTypeHints(),
+                FieldProvider.UNSUPPORTED,
+                null
+            );
+            ExpressionAnalysisContext exprCtx = new ExpressionAnalysisContext(context.sessionSettings());
+            HashMap<String, Symbol> options = HashMap.newHashMap(createServer.options().size());
+            for (var entry : createServer.options().entrySet()) {
+                String name = entry.getKey();
+                Expression value = entry.getValue();
+                options.put(name, expressionAnalyzer.convert(value, exprCtx));
+            }
+            return new AnalyzedCreateServer(
+                createServer.name(),
+                createServer.fdw(),
+                createServer.ifNotExists(),
+                options
+            );
+        }
+
+        @Override
+        public AnalyzedStatement visitCreateForeignTable(CreateForeignTable createForeignTable,
+                                                         Analysis context) {
+            RelationName tableName = RelationName.of(
+                createForeignTable.name(),
+                context.sessionSettings().searchPath().currentSchema()
+            );
+            tableName.ensureValidForRelationCreation();
+            var tableElementsAnalyzer = new TableElementsAnalyzer(
+                tableName,
+                context.transactionContext(),
+                nodeCtx,
+                context.paramTypeHints()
+            );
+            return tableElementsAnalyzer.analyze(createForeignTable);
+        }
+
+        @Override
+        public AnalyzedStatement visitCreateUserMapping(CreateUserMapping createUserMapping, Analysis context) {
+            String userName = createUserMapping.userName() == null
+                ? context.sessionSettings().userName()
+                : createUserMapping.userName();
+
+            Role user = roleManager.findUser(userName);
+            ExpressionAnalyzer expressionAnalyzer = new ExpressionAnalyzer(
+                context.transactionContext(),
+                nodeCtx,
+                context.paramTypeHints(),
+                FieldProvider.UNSUPPORTED,
+                null
+            );
+            ExpressionAnalysisContext exprCtx = new ExpressionAnalysisContext(context.sessionSettings());
+            HashMap<String, Symbol> options = HashMap.newHashMap(createUserMapping.options().size());
```

Comment:
This may look a bit duplicated from the create server options, but I kept it that way for now because I want to follow up on it and see if we can do some server/fdw specific validation logic.